From 0d9eb78d9d2fe6323c23b83e9d5f557defda6f31 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 11 Apr 2022 12:41:29 +0300
Subject: [PATCH] [SW-66329] module: pci: make run function dev specific

Pci run function can have different implementation for different
devices, for example pif arc device will implement the FLR handling
sequence as part of the pci mudule run function.

Change-Id: I725d903b76114af2aa3e8c2f6baaf88e7561302f
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/lib/common/cmd_module.h b/lib/common/cmd_module.h
index e9b138e..5111742 100644
--- a/lib/common/cmd_module.h
+++ b/lib/common/cmd_module.h
@@ -79,6 +79,7 @@
 void pci_dev_specific_enable(void);
 void pci_dev_specific_teardown(void);
 void pci_dev_specific_hot_rst(void);
+void pci_dev_specific_run(void);
 
 /* FWUPD */
 void module_fwupd_setup(void);
diff --git a/lib/preboot/module_pci.c b/lib/preboot/module_pci.c
index da56534..fe8124b 100644
--- a/lib/preboot/module_pci.c
+++ b/lib/preboot/module_pci.c
@@ -173,7 +173,7 @@
 	pci_sts.setup_done = false;
 }
 
-void module_pci_run(void)
+__weak void pci_dev_specific_run(void)
 {
 	unsigned int rst_src = hl_rst_src_get();
 
@@ -185,6 +185,11 @@
 	}
 }
 
+void module_pci_run(void)
+{
+	pci_dev_specific_run();
+}
+
 __weak void pci_dev_specific_post_link_up(void) {}
 
 void module_pci_run_fast(void)
